src/type/definition.js

```diff
@@ -757,11 +756,6 @@ function isPlainObj(obj) {
   return obj && typeof obj === 'object' && !Array.isArray(obj);
 }
 
-// If a resolver is defined, it must be a function.
-function isValidResolver(resolver: mixed): boolean {
```

Comment:
Should we keep `isValidResolver`? I'm not totally sure what the advantage of inlining it is

Reply:
One reason would be that you can't share it without changing the name since `isTypeOf` not a resolver. Plus it also confusing since `null` and `undefined` are not valid resolvers but an explicit absence of resolver.

So the best name that I come up is `isOptionalFunction` which is the same as to just write `resolver == null || typeof resolver === 'function'`.
I think it better to stay explicit.

Reply:
That's a reasonable argument. I'm convinced.